### ISSUE TITLE
Are directories excluded by .gitignore and friends supposed to be included in setup.py package_data?

### ISSUE THREAD

**Issue by aw1621107** [sdispater/poetry#431]
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://poetry.eustace.io/docs/) and believe that my question is not covered.

## Question

Should `poetry build` exclude things ignored by a VCS from the generated `package_data` in `setup.py`? I find it rather odd that having something in a `.gitignore` results in it not being included in the generated `.tar.gz`, but still having it mentioned in `setup.py`.

For example, I have `.mypy_cache/` ignored in my `.gitignore` files. They are correctly excluded from the generated `.tar.gz` file, but I end up with the following in `setup.py`:

```python3
package_data = \
{'': ['*'],
 'test': ['.mypy_cache/*',
          '.mypy_cache/3.7/*',
          '.mypy_cache/3.7/collections/*',
          '.mypy_cache/3.7/importlib/*',
          '.mypy_cache/3.7/os/*']}
```

Which, at least in my inexperienced eyes, indicates that `.mypy_cache/` and its contents are supposed to be installed, even though they aren't actually included in the source archive.

Is this intended? And if not, is there an easy way of avoiding this behavior besides just removing `.mypy_cache/` prior to running `poetry build`?

**Comment by nikolaiwo** [comment]
Maybe change `name, version = line.split("==")` to `name, version = re.split("={2,3}", line)`?

**Comment by sdispater** [comment]
No, excluded files and directories should not appear in the `setup.py`. This is definitely a bug, I will take a look at it.

**Comment by sdispater** [comment]
Could you provide the `pyproject.toml` file that generated this `setup.py` file?

**Comment by aw1621107** [comment]
Here's a minimal `pyproject.toml` that reproduces the issue for me:

```python3
[tool.poetry]
name = "test"
version = "0.1.0"
description = ""
authors = ["Alex Wang <aw1621107@gmail.com>"]

[tool.poetry.dependencies]
python = "^3.7"

[tool.poetry.dev-dependencies]
mypy = "^0.630.0"
```

The `.gitignore` I used was

```
.mypy_cache/
dist/
__pycache__/
```

I installed mypy through poetry, then created `test/test.py`:

```python3
def f(x: int) -> int:
    return x + 1

def g(x: int) -> int:
    return f("5")
```

Then `cd`'d to `test/` and ran `mypy test.py`, which emitted a type error as expected. Then I ran `poetry build`, and got that `setup.py`.

By the way, it seems that ignoring files in a `.gitignore` doesn't work unless there's an accompanying `.git/` folder. Is that intended?

**Comment by sdispater** [comment]
> By the way, it seems that ignoring files in a .gitignore doesn't work unless there's an accompanying .git/ folder. Is that intended?

Yes, this is intended since Poetry uses the `git` command internally to check for excluded files.
